python/lsst/jointcal/jointcalLib.i

```diff
@@ -101,16 +104,31 @@ namespace jointcal {
 }}
 
 %shared_ptr(lsst::jointcal::CcdImage);
-%template(CcdImageRef) boost::shared_ptr<lsst::jointcal::CcdImage>;
-%template(CcdImageRefList) std::list<boost::shared_ptr<lsst::jointcal::CcdImage> >;
-%template(TanSipPix2RaDecRef) boost::shared_ptr<lsst::jointcal::TanSipPix2RaDec>;
+// TODO: NOTE: this may not be needed? Unclear.
+// %template(CcdImageRef) std::shared_ptr<lsst::jointcal::CcdImage>;
+%template(CcdImageRefList) std::list<std::shared_ptr<lsst::jointcal::CcdImage> >;
+%template(TanSipPix2RaDecRef) std::shared_ptr<lsst::jointcal::TanSipPix2RaDec>;
+
+// %shared_ptr(lsst::jointcal::BaseStar);
+// %shared_ptr(lsst::jointcal::RefStar);
+// %shared_ptr(lsst::jointcal::FittedStar);
+// %template(RefStarRef) std::shared_ptr<lsst::jointcal::RefStar>;
+// %intrusive_ptr(lsst::jointcal::RefStar);
+// %template(StarListOfRefStar) std::list<std::intrusive_ptr<lsst::jointcal::RefStar> >;
+%include "lsst/jointcal/StarList.h"
+// %template(StarListMonkey) lsst::jointcal::StarList<lsst::jointcal::RefStar>;
+// typedef StarList std::list<lsst::jointcal::CountedRef>;
+// %template(RefStarRefList) std::list<lsst::jointcal::RefStar>;
```

Comment:
I see. You tried my earlier suggestion but didn't go with it for some reason (since this code is commented out). Did it not work? If so, I suggest adding those functions in this file instead of the .h file (since C++ doesn't need them). Or at the very least, explain in comments why they exist.